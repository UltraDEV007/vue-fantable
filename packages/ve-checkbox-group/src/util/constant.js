// prefix
export const PREFIX_CLS = 've-checkbox-group-'

// emit events
export const EMIT_EVENTS = {
  // on checked change
  ON_CHECKED_CHANGE: 'on-checked-change',
}

// comps name
export const COMPS_NAME = {
  VE_CHECKBOX: 'VeCheckbox',
  VE_CHECKBOX_GROUP: 'VeCheckboxGroup',
}
