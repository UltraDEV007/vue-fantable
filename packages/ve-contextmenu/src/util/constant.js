// prefix
export const PREFIX_CLS = 've-contextmenu-'

// comps name
export const COMPS_NAME = {
  VE_CONTEXTMENU: 'VeContextmenu',
}

// emit events
export const EMIT_EVENTS = {
  ON_NODE_CLICK: 'on-node-click',
}

// init data
export const INIT_DATA = {
  // PARENT_ID: 0,
  PARENT_DEEP: -1,
}

// contextmenu types
export const CONTEXTMENU_NODE_TYPES = {
  // Separator
  SEPARATOR: 'SEPARATOR',
}

// instance methods
export const INSTANCE_METHODS = {
  HIDE_CONTEXTMENU: 'hideContextmenu',
}
