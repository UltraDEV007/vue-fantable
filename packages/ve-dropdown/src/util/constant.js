// prefix
export const PREFIX_CLS = 've-dropdown-'

// emit events
export const EMIT_EVENTS = {
  FILTER_RESET: 'on-filter-reset',
  FILTER_CONFIRM: 'on-filter-confirm',
  VISIBLE_CHANGE: 'on-dropdown-visible-change',
  ITEM_SELECT_CHANGE: 'on-item-select-change',
}

// comps name
export const COMPS_NAME = {
  VE_DROPDOWN: 'VeDropdown',
}
