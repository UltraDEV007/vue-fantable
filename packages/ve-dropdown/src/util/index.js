/*
 * @clsName
 * @desc  get class name
 * @param {string} cls - class
 */
export function clsName(cls) {
  return 've-dropdown-' + cls
}
