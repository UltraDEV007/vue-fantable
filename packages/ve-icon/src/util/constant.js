// prefix
export const PREFIX_CLS = 've-icon-'

// emit events
export const EMIT_EVENTS = {}

// comps name
export const COMPS_NAME = {
  VE_ICON: 'VeIcon',
}
