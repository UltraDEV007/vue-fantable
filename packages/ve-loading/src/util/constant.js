// prefix
export const PREFIX_CLS = 've-loading-'

// emit events
export const EMIT_EVENTS = {}

// comps name
export const COMPS_NAME = {
  VE_LOADING: 'VeLoading',
  VE_LOADING_PLANE: 'VeLoadingPlane',
  /* VE_LOADING_CHASE: "VeLoadingChase", */
  VE_LOADING_BOUNCE: 'VeLoadingBounce',
  VE_LOADING_WAVE: 'VeLoadingWave',
  VE_LOADING_PULSE: 'VeLoadingPulse',
  VE_LOADING_FLOW: 'VeLoadingFlow',
  /* VE_LOADING_SWING: "VeLoadingSwing", */
  /*  VE_LOADING_CIRCLE: "VeLoadingCircle",
VE_LOADING_CIRCLE_FADE: "VeLoadingCircleFade", */
  VE_LOADING_GRID: 'VeLoadingGrid',
  VE_LOADING_FOLD: 'VeLoadingFold',
}

// spin name
export const SPIN_NAMES = {
  PLANE: 'plane',
  GRID: 'grid',
  WAVE: 'wave',
  FLOW: 'flow',
  BOUNCE: 'bounce',
  PULSE: 'pulse',
}
