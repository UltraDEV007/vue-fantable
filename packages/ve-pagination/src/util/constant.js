// prefix
export const PREFIX_CLS = 've-pagination-'

// locale comp name
export const LOCALE_COMP_NAME = 'pagination'

// emit events
export const EMIT_EVENTS = {
  PAGE_NUMBER_CHANGE: 'on-page-number-change',
  PAGE_SIZE_CHANGE: 'on-page-size-change',
}

// comps name
export const COMPS_NAME = {
  VE_PAGINATION: 'VePagination',
}
