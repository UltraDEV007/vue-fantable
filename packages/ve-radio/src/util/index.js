import { PREFIX_CLS } from './constant'

/*
 * @clsName
 * @desc  get class name
 * @param {string} cls - class
 */
export function clsName(cls) {
  return PREFIX_CLS + cls
}
