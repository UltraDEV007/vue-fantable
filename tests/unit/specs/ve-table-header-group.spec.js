import { mount } from "@vue/test-utils";
import veTable from "@/ve-table";
import { later } from "../util";

describe("veTable header group", () => {
    const TABLE_DATA = [
        {
            rowkey: 0,
            col1: "1",
            col2: "2",
            col3: "3",
            col4: "4",
            col5: "5",
            col6: "6",
            col7: "7",
            col8: "8",
            "col9-1": "9-1",
            "col9-2": "9-2",
            "col10-1": "10-1",
        },
        {
            rowkey: 1,
            col1: "1",
            col2: "2",
            col3: "3",
            col4: "4",
            col5: "5",
            col6: "6",
            col7: "7",
            col8: "8",
            "col9-1": "9-1",
            "col9-2": "9-2",
            "col10-1": "10-1",
        },
    ];

    const TABLE_COLUMNS = [
        {
            field: "col1",
            key: "a",
            title: "col1",
            width: "10%",
        },
        {
            title: "col2-col3",
            children: [
                {
                    field: "col2",
                    key: "b",
                    title: "col2",
                    width: 100,
                },
                {
                    field: "col3",
                    key: "c",
                    title: "col3",
                    width: 110,
                },
            ],
        },
        {
            title: "col4-col5-col6",
            children: [
                {
                    title: "col4-col5",
                    children: [
                        {
                            field: "col4",
                            key: "d",
                            title: "col4",
                            width: 130,
                        },
                        {
                            field: "col5",
                            key: "e",
                            title: "col5",
                            width: 140,
                        },
                    ],
                },
                {
                    title: "col6",
                    field: "col6",
                    key: "f",
                    width: 140,
                },
            ],
        },
        /*  {
            field: "col7",
            key: "g",
            title: "col7",
            width: 150
        },
        {
            field: "col8",
            key: "h",
            title: "col8",
            width: 160
        } */
    ];

    it("render", () => {
        const wrapper = mount(veTable, {
            propsData: {
                columns: TABLE_COLUMNS,
                tableData: TABLE_DATA,
                rowKeyFieldName: "rowkey",
            },
        });

        expect(wrapper.html()).toMatchSnapshot();
    });

    it("row last column", () => {
        const wrapper = mount(veTable, {
            propsData: {
                borderY: true,
                columns: TABLE_COLUMNS,
                tableData: TABLE_DATA,
                rowKeyFieldName: "rowkey",
            },
        });

        const trEls = wrapper.findAll(".ve-table-header-tr");

        // first row last column
        expect(
            trEls.at(0).findAll(".ve-table-header-th").at(2).classes(),
        ).toContain("ve-table-last-column");

        // second row last column
        expect(
            trEls.at(1).findAll(".ve-table-header-th").at(3).classes(),
        ).toContain("ve-table-last-column");
    });

    it("colspan and rowspan", () => {
        const wrapper = mount(veTable, {
            propsData: {
                columns: TABLE_COLUMNS,
                tableData: TABLE_DATA,
                rowKeyFieldName: "rowkey",
            },
        });

        const trEls = wrapper.findAll(".ve-table-header-tr");
        expect(trEls.length).toBe(3);

        /* first tr ths */
        const firstTrThEls = trEls.at(0).findAll(".ve-table-header-th");
        expect(firstTrThEls.length).toBe(3);
        expect(firstTrThEls.at(0).attributes("rowspan")).toBe("3");
        expect(firstTrThEls.at(0).attributes("colspan")).toBe("1");

        expect(firstTrThEls.at(1).attributes("rowspan")).toBe("1");
        expect(firstTrThEls.at(1).attributes("colspan")).toBe("2");

        expect(firstTrThEls.at(2).attributes("rowspan")).toBe("1");
        expect(firstTrThEls.at(2).attributes("colspan")).toBe("3");

        /* second tr ths */
        const secondTrThEls = trEls.at(1).findAll(".ve-table-header-th");
        expect(secondTrThEls.length).toBe(4);
        expect(secondTrThEls.at(0).attributes("rowspan")).toBe("2");
        expect(secondTrThEls.at(0).attributes("colspan")).toBe("1");

        expect(secondTrThEls.at(1).attributes("rowspan")).toBe("2");
        expect(secondTrThEls.at(1).attributes("colspan")).toBe("1");

        expect(secondTrThEls.at(2).attributes("rowspan")).toBe("1");
        expect(secondTrThEls.at(2).attributes("colspan")).toBe("2");

        expect(secondTrThEls.at(3).attributes("rowspan")).toBe("2");
        expect(secondTrThEls.at(3).attributes("colspan")).toBe("1");

        /* third tr ths */
        const thirdTrThEls = trEls.at(2).findAll(".ve-table-header-th");
        expect(thirdTrThEls.length).toBe(2);
        expect(thirdTrThEls.at(0).attributes("rowspan")).toBe("1");
        expect(thirdTrThEls.at(0).attributes("colspan")).toBe("1");

        expect(thirdTrThEls.at(1).attributes("rowspan")).toBe("1");
        expect(thirdTrThEls.at(1).attributes("colspan")).toBe("1");
    });

    it("width right column fixed section1", () => {
        /**
         * +------------------------------+
         * |        | Title2   | Title3   |
         * | Title1 |----------|----------|
         * |        | Title2-1 | Title3-1 |
         * +------------------------------+
         */

        const wrapper = mount(veTable, {
            propsData: {
                borderY: true,
                columns: [
                    { field: "col1", key: "a", title: "Title1" },
                    {
                        title: "Title2",
                        fixed: "right",
                        key: "b",
                        children: [
                            {
                                field: "col2-1",
                                key: "b-1",
                                title: "Title2-1",
                                width: 200,
                            },
                        ],
                    },
                    {
                        title: "Title3",
                        fixed: "right",
                        key: "c",
                        children: [
                            {
                                field: "col10-1",
                                key: "c-1",
                                title: "Title3-1",
                                width: 200,
                            },
                        ],
                    },
                ],
                tableData: TABLE_DATA,
                rowKeyFieldName: "rowkey",
            },
        });

        const trEls = wrapper.findAll(".ve-table-header-tr");

        // first row first fixed column
        const firtRowFirstFixedColumn = trEls
            .at(0)
            .findAll(".ve-table-header-th")
            .at(1);

        expect(firtRowFirstFixedColumn.classes()).toEqual([
            "ve-table-header-th",
            "ve-table-fixed-right",
            "ve-table-first-right-fixed-column",
        ]);
        expect(firtRowFirstFixedColumn.attributes("style")).toContain(
            "right: 200px;",
        );

        // first row last fixed column
        const firstRowLastFixedColumn = trEls
            .at(0)
            .findAll(".ve-table-header-th")
            .at(2);

        expect(firstRowLastFixedColumn.classes()).toEqual([
            "ve-table-header-th",
            "ve-table-fixed-right",
            "ve-table-last-column",
        ]);
        expect(firstRowLastFixedColumn.attributes("style")).toContain(
            "right: 0px;",
        );

        // second row first fixed column
        const secondRowFirstFixedColumn = trEls
            .at(1)
            .findAll(".ve-table-header-th")
            .at(0);

        expect(secondRowFirstFixedColumn.classes()).toEqual([
            "ve-table-header-th",
            "ve-table-fixed-right",
            "ve-table-first-right-fixed-column",
        ]);
        expect(secondRowFirstFixedColumn.attributes("style")).toContain(
            "right: 200px;",
        );

        // second row last fixed column
        const secondRowLastFixedColumn = trEls
            .at(1)
            .findAll(".ve-table-header-th")
            .at(1);

        expect(secondRowLastFixedColumn.classes()).toEqual([
            "ve-table-header-th",
            "ve-table-fixed-right",
            "ve-table-last-column",
        ]);
        expect(secondRowLastFixedColumn.attributes("style")).toContain(
            "right: 0px;",
        );
    });

    it("width right column fixed section2", () => {
        /**
         * +------------------------------+
         * |        |       Title2        |
         * | Title1 |----------|----------|
         * |        | Title2-1 | Title2-2 |
         * +------------------------------+
         */

        const wrapper = mount(veTable, {
            propsData: {
                borderY: true,
                columns: [
                    { field: "col1", key: "a", title: "Title1" },
                    {
                        title: "Title2",
                        fixed: "right",
                        key: "b",
                        children: [
                            {
                                field: "col9-1",
                                key: "b-1",
                                title: "Title2-1",
                                width: 200,
                            },
                            {
                                field: "col9-2",
                                key: "b-2",
                                title: "Title2-2",
                                width: 200,
                            },
                        ],
                    },
                ],
                tableData: TABLE_DATA,
                rowKeyFieldName: "rowkey",
            },
        });

        const trEls = wrapper.findAll(".ve-table-header-tr");

        // first row first fixed column
        const firtRowFirstFixedColumn = trEls
            .at(0)
            .findAll(".ve-table-header-th")
            .at(1);

        expect(firtRowFirstFixedColumn.classes()).toEqual([
            "ve-table-header-th",
            "ve-table-fixed-right",
            "ve-table-first-right-fixed-column",
            "ve-table-last-column",
        ]);
        expect(firtRowFirstFixedColumn.attributes("style")).toContain(
            "right: 0px;",
        );

        // second row first fixed column
        const secondRowFirstFixedColumn = trEls
            .at(1)
            .findAll(".ve-table-header-th")
            .at(0);

        expect(secondRowFirstFixedColumn.classes()).toEqual([
            "ve-table-header-th",
            "ve-table-fixed-right",
            "ve-table-first-right-fixed-column",
        ]);
        expect(secondRowFirstFixedColumn.attributes("style")).toContain(
            "right: 200px;",
        );

        // second row last fixed column
        const secondRowLastFixedColumn = trEls
            .at(1)
            .findAll(".ve-table-header-th")
            .at(1);

        expect(secondRowLastFixedColumn.classes()).toEqual([
            "ve-table-header-th",
            "ve-table-fixed-right",
            "ve-table-last-column",
        ]);
        expect(secondRowLastFixedColumn.attributes("style")).toContain(
            "right: 0px;",
        );
    });
});
