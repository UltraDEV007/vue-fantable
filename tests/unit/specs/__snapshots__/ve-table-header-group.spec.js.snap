// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`veTable header group render 1`] = `
<div class="vue-table-root">
  <div class="fan-table fan-table-border-around" style="width: 100%;">
    <div class="fan-table-container fan-table-enable-cell-selection" style="height: 0px;">
      <div class="fan-table-content-wrapper">
        <table class="fan-table-content fan-table-border-x">
          <colgroup>
            <col style="width: 10%;">
            <col style="width: 100px;">
            <col style="width: 110px;">
            <col style="width: 130px;">
            <col style="width: 140px;">
            <col style="width: 140px;">
          </colgroup>
          <thead class="fan-table-fixed-header fan-table-header">
            <tr class="fan-table-header-tr">
              <th rowspan="3" colspan="1" class="fan-table-header-th" style="text-align: center; top: 0px;">col1</th>
              <th rowspan="1" colspan="2" class="fan-table-header-th" style="text-align: center; top: 0px;">col2-col3</th>
              <th rowspan="1" colspan="3" class="fan-table-header-th fan-table-last-column" style="text-align: center; top: 0px;">col4-col5-col6</th>
            </tr>
            <tr class="fan-table-header-tr">
              <th rowspan="2" colspan="1" class="fan-table-header-th" style="text-align: center; top: 0px;">col2</th>
              <th rowspan="2" colspan="1" class="fan-table-header-th" style="text-align: center; top: 0px;">col3</th>
              <th rowspan="1" colspan="2" class="fan-table-header-th" style="text-align: center; top: 0px;">col4-col5</th>
              <th rowspan="2" colspan="1" class="fan-table-header-th fan-table-last-column" style="text-align: center; top: 0px;">col6</th>
            </tr>
            <tr class="fan-table-header-tr">
              <th rowspan="1" colspan="1" class="fan-table-header-th" style="text-align: center; top: 0px;">col4</th>
              <th rowspan="1" colspan="1" class="fan-table-header-th" style="text-align: center; top: 0px;">col5</th>
            </tr>
          </thead>
          <tbody class="fan-table-body fan-table-row-hover fan-table-row-highlight">
            <tr style="height: 0px;">
              <td style="padding: 0px; border: 0px; height: 0px;"></td>
              <td style="padding: 0px; border: 0px; height: 0px;"></td>
              <td style="padding: 0px; border: 0px; height: 0px;"></td>
              <td style="padding: 0px; border: 0px; height: 0px;"></td>
              <td style="padding: 0px; border: 0px; height: 0px;"></td>
              <td style="padding: 0px; border: 0px; height: 0px;"></td>
            </tr>
            <tr row-key="0" class="fan-table-body-tr">
              <td rowspan="1" colspan="1" col-key="a" class="fan-table-body-td" style="text-align: center;">1</td>
              <td rowspan="1" colspan="1" col-key="b" class="fan-table-body-td" style="text-align: center;">2</td>
              <td rowspan="1" colspan="1" col-key="c" class="fan-table-body-td" style="text-align: center;">3</td>
              <td rowspan="1" colspan="1" col-key="d" class="fan-table-body-td" style="text-align: center;">4</td>
              <td rowspan="1" colspan="1" col-key="e" class="fan-table-body-td" style="text-align: center;">5</td>
              <td rowspan="1" colspan="1" col-key="f" class="fan-table-body-td" style="text-align: center;">6</td>
            </tr>
            <tr row-key="1" class="fan-table-body-tr">
              <td rowspan="1" colspan="1" col-key="a" class="fan-table-body-td" style="text-align: center;">1</td>
              <td rowspan="1" colspan="1" col-key="b" class="fan-table-body-td" style="text-align: center;">2</td>
              <td rowspan="1" colspan="1" col-key="c" class="fan-table-body-td" style="text-align: center;">3</td>
              <td rowspan="1" colspan="1" col-key="d" class="fan-table-body-td" style="text-align: center;">4</td>
              <td rowspan="1" colspan="1" col-key="e" class="fan-table-body-td" style="text-align: center;">5</td>
              <td rowspan="1" colspan="1" col-key="f" class="fan-table-body-td" style="text-align: center;">6</td>
            </tr>
          </tbody>
          <tfoot class="fan-table-fixed-footer fan-table-footer"></tfoot>
        </table>
        <div class="fan-table-selection-wrapper">
          <div class="fan-table-selection-fixed-left"></div>
          <div class="fan-table-selection-middle"></div>
          <div class="fan-table-selection-fixed-right"></div>
        </div>
      </div>
    </div>
    <div class="fan-table-edit-input-container" style="top: 0px; left: 0px; height: 1px; z-index: -1; opacity: 0;"><textarea tabindex="-1" class="fan-table-edit-input"></textarea></div>
  </div>
</div>
`;
